src/lib/schematics/ng-update/upgrade-rules/package-imports-v8/updateAngularMaterialImportsRule.ts

```diff
@@ -0,0 +1,172 @@
+      return;
+    }
+    const importLocation = declaration.moduleSpecifier.getText(sf);
+    // If the import module is not @angular/material, skip check.
+    if (importLocation !== ANGULAR_MATERIAL_IMPORT_PATH) {
+      return;
+    }
+
+    // If no import clause is found, or nothing is named as a binding in the
+    // import, add failure saying to import symbols in clause.
+    if (!declaration.importClause || !declaration.importClause.namedBindings) {
+      return ctx.addFailureAtNode(declaration, Rule.NO_IMPORT_NAMED_SYMBOLS_FAILURE_STR);
+    }
+
+    // All named bindings in import clauses must be named symbols, otherwise add
+    // failure saying to import symbols in clause.
+    if (!ts.isNamedImports(declaration.importClause.namedBindings)) {
+      return ctx.addFailureAtNode(declaration, Rule.NO_IMPORT_NAMED_SYMBOLS_FAILURE_STR);
+    }
+
+    // If no symbols are in the named bindings then add failure saying to
+    // import symbols in clause.
+    if (!declaration.importClause.namedBindings.elements.length) {
+      return ctx.addFailureAtNode(declaration, Rule.NO_IMPORT_NAMED_SYMBOLS_FAILURE_STR);
+    }
+
+    // Map of submodule locations to arrays of imported symbols.
+    const importMap = new Map<string, Set<string>>();
+
+    // Determine the subpackage each symbol in the namedBinding comes from.
+    for (const importedSymbol of declaration.importClause.namedBindings.elements) {
+      // Confirm the named import is a symbol that can be looked up.
+      if (!ts.isIdentifier(importedSymbol.name)) {
+        return ctx.addFailureAtNode(
+            importedSymbol, importedSymbol.getFullText(sf) + Rule.SYMBOL_NOT_FOUND_FAILURE_STR);
+      }
+      // Get the original symbol where it is declared upstream.
+      const symbol = getOriginalSymbol(importedSymbol.name, checker);
+
+      // If the symbol can't be found, add failure saying the symbol
+      // can't be found.
+      if (!symbol) {
+        return ctx.addFailureAtNode(
+            importedSymbol, importedSymbol.getFullText(sf) + Rule.SYMBOL_NOT_FOUND_FAILURE_STR);
+      }
+
+      // If the symbol has no declarations, add failure saying the symbol can't
+      // be found.
+      if (!symbol.declarations || !symbol.declarations.length) {
+        return ctx.addFailureAtNode(
+            importedSymbol, importedSymbol.getFullText(sf) + Rule.SYMBOL_NOT_FOUND_FAILURE_STR);
+      }
+
+      // The filename for the source file of the node that contains the
+      // first declaration of the symbol.  All symbol declarations must be
+      // part of a defining node, so parent can be asserted to be defined.
+      const fileName = symbol.declarations[0].parent.getSourceFile().fileName;
+      // File the module the symbol belongs to from a regex match of the
+      // filename. This will always match since only @angular/material symbols
+      // are being looked at.
+      const matched = fileName.match(ANGULAR_MATERIAL_FILEPATH_REGEX)!;
```

Comment:
Nit: use destructuring and also check if there is a match before adding it to the map?